specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-10-01-preview/BulkActions.json

```diff
@@ -0,0 +1,652 @@
+            "type": "string"
+          },
+          "mutator": {
+            "description": "Mutator for this operation step.",
+            "type": "string",
+            "enum": [
+              "SetValue",
+              "SetNull",
+              "SetEmpty",
+              "AddToSet",
+              "AppendToList",
+              "RemoveFromSet"
+            ],
+            "x-ms-enum": {
+              "modelAsString": true,
+              "name": "OperationMutator",
+              "values": [
+                {
+                  "description": "Use this operator when a scalar value should be replaced/overwritten.",
+                  "value": "SetValue"
+                },
+                {
+                  "description": "Use this operator when a scalar or array value should be set explicitly to NULL.",
+                  "value": "SetNull"
+                },
+                {
+                  "description": "Use this operator when an array value should be set to the empty set/list.",
+                  "value": "SetEmpty"
+                },
+                {
+                  "description": "Use this operator to add an element to a set (i.e. an unordered multivalue variable).",
+                  "value": "AddToSet"
+                },
+                {
+                  "description": "Use this operator to add an element to the end of a list (i.e. an ordered multivalue variable).",
+                  "value": "AppendToList"
+                },
+                {
+                  "description": "Use this operator to remove the specified element from a set (i.e. an unordered multivalue variable).",
+                  "value": "RemoveFromSet"
+                }
+              ]
+            }
+          },
+          "value": {
+            "description": "Value to be applied by this operation step.",
+            "type": "string"
+          }
+        }    
+      },
+      "Operation": {
+        "description": "Represents the operation to be applied to all items captured by this bulk action.",
+        "type": "object",
+        "properties": {
+          "type": {
+            "description": "Describes which type of operation we are applying to items captured by this bulk action.",
+            "type": "string",
+            "enum": [
+              "Delete",
+              "Edit"
+            ],
+            "x-ms-enum": {
+              "modelAsString": true,
+              "name": "OperationType",
+              "values": [
+                {
+                  "description": "This item will be deleted.",
+                  "value": "Delete"
+                },
+                {
+                  "description": "This item will be edited.",
+                  "value": "Edit"
+                }
+              ]
+            }
+          },
+          "steps": {
+            "description": "List of operation steps to be applied.",
+            "type": "array",
```

Comment:
is this intended to be an ordered list of steps? If so you can introduce a property to let customers define the order explicitly. If the index of the array is being used to define the order then please document it as such. If this is not an ordered list, pl declare that as well in the documentation